buildSrc/src/main/java/org/elasticsearch/gradle/release/BreakingChangesGenerator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.release;
+
+import org.elasticsearch.gradle.Version;
+import org.elasticsearch.gradle.VersionProperties;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+public class BreakingChangesGenerator implements Closeable {
+
+    private final PrintStream out;
+
+    public BreakingChangesGenerator(File outputFile) throws FileNotFoundException {
+        this.out = new PrintStream(outputFile);
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.out.close();
+    }
+
+    public void generate(List<ChangelogEntry> entries) {
+        final Version version = VersionProperties.getElasticsearchVersion();
+
+        String majorDotMinor = version.getMajor() + "." + version.getMinor();
+        String majorMinor = String.valueOf(version.getMajor()) + version.getMinor();
+
+        List.of(
+            "[[breaking-changes-" + majorDotMinor + "]]",
+            "== Breaking changes in " + majorDotMinor,
+            "++++",
+            "<titleabbrev>" + majorDotMinor + "</titleabbrev>",
+            "++++",
+            "",
+            "This section discusses the changes that you need to be aware of when migrating",
+            "your application to {es} " + majorDotMinor + ".",
+            "",
+            "See also <<release-highlights>> and <<es-release-notes>>."
+        ).forEach(out::println);
+
+        if (VersionProperties.isElasticsearchSnapshot()) {
+            out.println();
+            out.println("coming[" + version + "]");
+        }
+
+        out.println();
+        out.println("//NOTE: The notable-breaking-changes tagged regions are re-used in the");
+        out.println("//Installation and Upgrade Guide");
```

Comment:
It doesn't look like the change has been applied to 7.11, 7.x, or master yet, but we've changed the format of our breaking changes to a migration guide:
https://www.elastic.co/guide/en/elasticsearch/reference/7.11/migrating-7.10.html

I think we'll need to update this file to separate breaking changes and deprecations. We'll also need to update the boilerplate text here.

Reply:
@debadair authored the migration guide changes. I'll add her as a reviewer to get her feedback.